.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs

```diff
@@ -287,6 +287,16 @@ private List<Token> NumberWithMonth(string text, DateObject reference)
 
             foreach (var result in er)
             {
+                // Check that the extracted number is not part of a decimal number (e.g. 123.24)
+                var beforeStr = text.Substring(0, (int)result.Start);

```

Comment:
Instead of allocating new objects, can't we read the previous char and make two comparisons?

Reply:
ok, modified